domain/batch.js

```diff
@@ -45,6 +53,9 @@ class BatchFile {
         this.filePrefix = filePrefix;
         this.createdAt = this.updatedAt = getCurrentTime();
         this.errors = [];
+        if (isDataFile){
```

Comment:
If isDatafile is false, does that mean we dont intend to create any nodeType?

Reply:
If it is not a data file, then the nodeType is added during validation because it cannot be determined until the file is parsed.